origo/pipelines/resources/pipeline_instance.py

```diff
@@ -65,6 +65,8 @@ def __dict__(self):
             "taskConfig": self.taskConfig,
             "useLatestEdition": self.useLatestEdition,
         }
+        if self.taskConfig is not None:
```

Comment:
Should we do just `if self.taskConfig`, i.e. omit it also if it is `{}`?

Reply:
Not sure, but I feel like it's cleanest like this, since it matches the behavior of the other optional fields (even though we currently treat the semantics of "missing config" and "empty config" the same in the pipeline processor).

Reply:
I'm fine with that, but then we should also add some validation that e.g. `pipelineArn` and `pipelineProcessordId` cannot be empty strings?

Reply:
Also, `schemaId` and `useLatestEdition` are no longer in use AFAIK, but can remove them in separate PR I guess 😀

Reply:
> I'm fine with that, but then we should also add some validation that e.g. pipelineArn and pipelineProcessordId cannot be empty strings?

Unsure how much validation we should do in this (currently) "dumb" data class. `""` is just one of infinitely many invalid values for `pipelineArn`/`pipelineProcessordId`, but do we really want to start doing data validation in here?